site/pages/token-revokes.js

```diff
@@ -0,0 +1,359 @@
+      if (!active) {
+        return
+      }
+
+      const subscription = library.eth.subscribe('logs', {
+        topics: [
+          APPROVAL_TOPIC,
+          library.utils.padLeft(account.toLowerCase(), 64)
+        ]
+      })
+
+      subscription.on('data', function (log) {
+        setSyncBlock((prev) => ({
+          ...prev,
+          tokenApprovals: getNewestApprovals({
+            library,
+            logs: parseLogs([log]),
+            tokenApprovals: prev.tokenApprovals
+          })
+        }))
+      })
+
+      return function () {
+        subscription.unsubscribe()
+      }
+    },
+    [active, library, account, chainId, setSyncBlock]
+  )
+
+  return { ...syncBlock, syncStatus: syncStatus, setSyncStatus }
+}
+
+const useErc20Token = function (address) {
+  const { active, library } = useWeb3React()
+  return useSWR(active ? address : null, function () {
+    const erc20Service = createErc20(library, address)
+    return Promise.all([
+      erc20Service.symbol(),
+      erc20Service.decimals(),
+      erc20Service.totalSupply()
+    ])
+  })
+}
+
+const Allowance = function ({ address, data }) {
+  const { library } = useWeb3React()
+  const { t } = useTranslation('common')
+
+  const { data: token } = useErc20Token(address)
+
+  if (!token) {
+    return <span className="m-auto"></span>
+  }
+  const [, decimals, totalSupply] = token
+  const allowanceInWei = library.utils.hexToNumberString(data)
+  const value = toFixed(fromUnit(allowanceInWei, decimals), 6)
+  const isUnlimited = Big(totalSupply).times(10).lt(allowanceInWei)
+  return <span className="m-auto">{isUnlimited ? t('unlimited') : value}</span>
+}
+
+const Token = function ({ address }) {
+  const { data: token } = useErc20Token(address)
+  if (!token) {
+    return <span className="m-auto"></span>
+  }
+  const [symbol] = token
+  return <span className="m-auto">{symbol}</span>
+}
+
+const TokenRevokes = function () {
+  const { active } = useWeb3React()
+  const { erc20 } = useContext(PureContext)
+  const { t } = useTranslation('common')
+  const [isRevoking, setIsRevoking] = useState(false)
+
+  const { tokenApprovals, syncStatus, setSyncStatus } = useTokenApprovals()
+
+  const handleRevoke = function (address, spender) {
+    if (!active) {
+      return
+    }
+    setIsRevoking(true)
+    const erc20Service = erc20(address)
+    erc20Service
+      .revoke(spender)
+      .then(() => setIsRevoking(false))
+      .catch(() => setSyncStatus(SyncStatus.Error))
+  }
+
+  return (
+    <Layout walletConnection>
+      {!active && <h3>{t('connect-to-sync')}</h3>}
+      {active && syncStatus === SyncStatus.Syncing && (
+        <h3>{t('syncing-your-approvals')}</h3>
+      )}
+      {active && isRevoking && <h3>{t('revoking-approval')}</h3>}
+      {syncStatus === SyncStatus.Error && <h3>{t('generic-error')}</h3>}
+      {tokenApprovals.length === 0 && syncStatus === SyncStatus.Finished && (
+        <p>{t('no-approvals')}</p>
+      )}
```

Comment:
These seems to be the "notification area". In other mini-apps, this appears at the bottom. Let's place it below the table.